frontend/src/features/documents/documentDetail/hooks/useDocumentProvider.ts

```diff
@@ -0,0 +1,130 @@
+    loading: downloadDocumentFileLoading,
+  } = useApiRequestWrapper<
+    (
+      mayanDocumentId: number,
+      mayanFileId: number,
+    ) => Promise<AxiosResponse<ExternalResult<FileDownload>, any>>
+  >({
+    requestFunction: useCallback(
+      async (mayanDocumentId: number, mayanFileId: number) =>
+        await downloadDocumentFileApiCall(mayanDocumentId, mayanFileId),
+      [downloadDocumentFileApiCall],
+    ),
+    requestName: 'DownloadDocumentFile',
+    onSuccess: useCallback(() => toast.success('Download Downloaded'), []),
+    onError: useCallback((axiosError: AxiosError<IApiError>) => {
+      if (axiosError?.response?.status === 400) {
+        toast.error(axiosError?.response.data.error);
+      } else {
+        toast.error('Download document error. Check responses and try again.');
+      }
+    }, []),
+  });
+
+  // Provides functionality for download the latest file for a document
+  const {
+    execute: downloadDocumentFileLatest,
+    loading: downloadDocumentFileLatestLoading,
+  } = useApiRequestWrapper<
+    (documendId: number) => Promise<AxiosResponse<ExternalResult<FileDownload>, any>>
+  >({
+    requestFunction: useCallback(
+      async (mayanDocumentId: number) => await downloadDocumentFileLatestApiCall(mayanDocumentId),
+      [downloadDocumentFileLatestApiCall],
+    ),
+    requestName: 'DownloadDocumentFileLatest',
+    onSuccess: useCallback(() => toast.success('Download Downloaded Latest'), []),
+    onError: useCallback((axiosError: AxiosError<IApiError>) => {
+      if (axiosError?.response?.status === 400) {
+        toast.error(axiosError?.response.data.error);
+      } else {
+        toast.error('Download document latest error. Check responses and try again.');
+      }
+    }, []),
+  });
+
+  // Provides functionality for deleting a document
+  const {
+    execute: deleteActivityDocument,
+    loading: deleteActivityDocumentLoading,
+  } = useApiRequestWrapper<
+    (documendId: number, activityId: number) => Promise<AxiosResponse<boolean, any>>
+  >({
+    requestFunction: useCallback(
+      async (documentId: number, activityId: number) =>
+        await deleteDocumentActivityApiCall(documentId, activityId),
+      [deleteDocumentActivityApiCall],
+    ),
+    requestName: 'deleteActivityDocument',
+    onSuccess: useCallback(() => toast.success('Delete activity document'), []),
+    onError: useCallback((axiosError: AxiosError<IApiError>) => {
+      if (axiosError?.response?.status === 400) {
+        toast.error(axiosError?.response.data.error);
+      } else {
+        toast.error('Delete activity document error. Check responses and try again.');
+      }
+    }, []),
+  });
+
+  return {
+    retrieveDocumentMetadata,
+    retrieveDocumentMetadataLoading,
+    downloadDocumentFile,
+    downloadDocumentFileLoading,
+    downloadDocumentFileLatest,
+    downloadDocumentFileLatestLoading,
+    deleteActivityDocument,
+    deleteActivityDocumentLoading,
```

Comment:
nit: we should also return the error state for consumers of this hook to check for errors (if they want to). We should discuss on Friday how to achieve that (if we want)

I've been returning the whole object from "useApiRequestWrapper" in my hooks.  For example:

```tsx
const retrieveDocumentMetadataApi =  useApiRequestWrapper(...)


// export
return {
  retrieveDocumentMetadata: retrieveDocumentMetadataApi

}

/// in a consumer of this hook

const { retrieveDocumentMetadata } = useDocumentProvider()
...
retrieveDocumentMetadata.execute(...)

retrieveDocumentMetadata.loading etc
```